### PR TITLE
Zabz dust

### PR DESCRIPTION
**Pull request by zabzzz**
main pr from dust pid testing

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -28,35 +28,58 @@
...
-    public static final double PIVOT_DOWN_SPEED = -0.2;
+    public static final double PIVOT_UP_SPEED = 0.1;
+    public static final double PIVOT_DOWN_SPEED = -0.1;
     public static final double MOTOR_ZERO_SPEED = 0;
+    public static final double MOTOR_GRAV_SPEED = 0.02;
+    public static final double MOTOR_LOADBACK_SPEED = -0.07;
+    public static final double MOTOR_LOADING_SPEED = 0.07;
 
     // Encoder Values for the Intake and Pivot
-    public static final double PIVOT_MAX = 10;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/Constants.java:50]: break out this comment into another variable

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -28,35 +28,58 @@
...
+    public static final double PIVOT_MAX = -6; // -8.5 for 5 FEET score
     public static final double PIVOT_MIN = 0;
 
     public static final int MOTOR_CURRENT_LIMIT = 40;
   }
 
+  public static final class SensorConstants {
+    
+    public static final int sensorPort1 = 0;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/Constants.java]: what is sensorPort1? is there a sensorPort0? if not rename this variable to something more sensible

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -47,6 +52,7 @@ public class RobotContainer {
   // private AutoBuilder autoBuilder = new AutoBuilder(swerveDrive,
   // intakeSubsystem, pivotSubsystem);
   private final SendableChooser<Command> autoChooser;
+  private double testMotorSpeeds;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/RobotContainer.java]: remove

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -60,6 +66,7 @@ public RobotContainer() {
     // DriverStation.startDataLog(DataLogManager.getLog(), false);
     // limeSubsystem.enableVisionProcessing();
     // Configure the button bindings
+    testMotorSpeeds = 0;
```
Comment (minhd-vu) [inline src/main/java/frc/robot/RobotContainer.java]: remove

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -105,22 +112,22 @@ public static double getDriverRightX() {
...
-    m_coDriverController.button(OIConstants.X_BUTTON).onTrue(new PivotUp(IntakeConstants.PIVOT_UP_SPEED));
+    m_driverController.rightTrigger().whileTrue(new Shooter(ManipulatorConstants.SCORE_HIGH_SPEED));
+    m_driverController.button(OIConstants.RIGHT_BUMPER).whileTrue(new Shooter(ManipulatorConstants.SCORE_MID_SPEED));
+    m_driverController.button(OIConstants.LEFT_BUMPER).whileTrue(new Shooter(ManipulatorConstants.SCORE_LOW_SPEED));
+    m_driverController.leftTrigger().whileTrue(new Intake(ManipulatorConstants.INTAKE_SPEED));
+    m_driverController.button(OIConstants.A_BUTTON).whileTrue(new PivotPIDCommand(ManipulatorConstants.PIVOT_MIN));
+    m_driverController.button(OIConstants.X_BUTTON).whileTrue(new PivotPIDCommand(ManipulatorConstants.PIVOT_MAX));
+    m_driverController.button(OIConstants.Y_BUTTON).whileTrue(new Feeder());
+
+    // // m_coDriverController.button(OIConstants.RIGHT_STICK_PRESS).whileTrue(new
```
Comment (minhd-vu) [inline src/main/java/frc/robot/RobotContainer.java]: remove these comments if you're not goign to sue them

File: src/main/java/frc/robot/commands/manipulator/Intake.java [hunk 6]
```diff
@@ -1,12 +1,13 @@
-// Copyright (c) FIRST and other WPILib contributors.
-// Open Source Software; you can modify and/or share it under the terms of
-// the WPILib BSD license file in the root directory of this project.
+// // Copyright (c) FIRST and other WPILib contributors.
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/Intake.java]: these are doulbe commented fix this

File: src/main/java/frc/robot/commands/manipulator/Shooter.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  public void initialize() {
+    commandTimer.reset();
+    commandTimer.start();
+  }
+
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    RobotContainer.shooterSubsystem.set(shootSpeed);
+    if (commandTimer.get() > 2.5) {
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/Shooter.java]: this should be a variable

File: src/main/java/frc/robot/commands/manipulator/Shooter.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  @Override
+  public void end(boolean interrupted) {
+    RobotContainer.shooterSubsystem.set(ManipulatorConstants.MOTOR_ZERO_SPEED);
+    RobotContainer.intakeSubsystem.set(ManipulatorConstants.INTAKE_REST_SPEED);
+  }
+
+  // Returns true when the command should end.
+  @Override
+  public boolean isFinished() {
+    if (commandTimer.get() > 4){
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/Shooter.java]: just do `return commandTimer.get() > 4` it should also be a variable

File: src/main/java/frc/robot/commands/manipulator/pivot/PivotDown.java [hunk 9]
```diff
@@ -1,12 +1,12 @@
-// Copyright (c) FIRST and other WPILib contributors.
-// Open Source Software; you can modify and/or share it under the terms of
-// the WPILib BSD license file in the root directory of this project.
+// // Copyright (c) FIRST and other WPILib contributors.
+// // Open Source Software; you can modify and/or share it under the terms of
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/pivot/PivotDown.java:2]: remove double commetns

File: src/main/java/frc/robot/commands/manipulator/pivot/PivotDown.java [hunk 10]
```diff
@@ -40,28 +40,19 @@ public void initialize() {
   @Override
   public void execute() {
 
-    if (Math.abs(RobotContainer.pivotSubsystem.getPivotMotorHeight()) < 18 ) {
-      RobotContainer.pivotSubsystem.set(-pivotSpeed);
-
+    if (Math.abs(RobotContainer.pivotSubsystem.getPivotMotorLHeight()) > 0 ) { //18 = pivot min height i think
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/pivot/PivotDown.java]: remove comment and place in a variable if you need it

File: src/main/java/frc/robot/commands/manipulator/pivot/PivotPIDCommand.java [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    RobotContainer.pivotSubsystem.setSetpoint(setpoint);
+  }
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    // RobotContainer.pivotSubsystem.set(ManipulatorConstants.MOTOR_GRAV_SPEED);
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/pivot/PivotPIDCommand.java]: remove comment

File: src/main/java/frc/robot/commands/manipulator/pivot/PivotUp.java [hunk 12]
```diff
@@ -1,12 +1,12 @@
-// Copyright (c) FIRST and other WPILib contributors.
-// Open Source Software; you can modify and/or share it under the terms of
-// the WPILib BSD license file in the root directory of this project.
+// // Copyright (c) FIRST and other WPILib contributors.
+// // Open Source Software; you can modify and/or share it under the terms of
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/pivot/PivotUp.java:2]: remove double comment

File: src/main/java/frc/robot/commands/manipulator/pivot/PivotUp.java [hunk 13]
```diff
@@ -40,28 +40,19 @@ public void initialize() {
   @Override
   public void execute() {
 
-    if (Math.abs(RobotContainer.pivotSubsystem.getPivotMotorHeight()) > 2 ) {
-      RobotContainer.pivotSubsystem.set(-pivotSpeed);
-
+    if (Math.abs(RobotContainer.pivotSubsystem.getPivotMotorLHeight()) < 10.9) { // 2 = pivot max height I think
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/manipulator/pivot/PivotUp.java]: remove comemnt, place in variable if needed

File: src/main/java/frc/robot/commands/vision/LimeAlign.java [hunk 14]
```diff
@@ -1,37 +1,37 @@
-// Copyright (c) FIRST and other WPILib contributors.
-// Open Source Software; you can modify and/or share it under the terms of
-// the WPILib BSD license file in the root directory of this project.
+// // Copyright (c) FIRST and other WPILib contributors.
+// // Open Source Software; you can modify and/or share it under the terms of
```
Comment (minhd-vu) [inline src/main/java/frc/robot/commands/vision/LimeAlign.java]: remove double coments

Comment (minhd-vu) [reply src/main/java/frc/robot/commands/vision/LimeAlign.java]: or remove entire file if not using

File: src/main/java/frc/robot/subsystems/DriveSubsystem.java [hunk 15]
```diff
@@ -116,7 +116,7 @@ public void periodic() {
             m_rearLeft.getPosition(),
             m_rearRight.getPosition()
         });
-    m_field.setRobotPose(m_odometry.getPoseMeters());
+    // m_field.setRobotPose(m_odometry.getPoseMeters());
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/DriveSubsystem.java:119]: remove if not using

File: src/main/java/frc/robot/subsystems/PivotSubsystem.java [hunk 16]
```diff
@@ -6,54 +6,92 @@
...
+import frc.robot.RobotContainer;
+import edu.wpi.first.math.controller.PIDController;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 
 /**
  * The PivotSubsystem has the motor objects for the motors of the
  * Pivot on the robot. It also sets them a value based on the input
  * received from a command
- * - 
+ * -
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/PivotSubsystem.java:23]: fix these comments, ith ink the formatting may be wiered

File: src/main/java/frc/robot/subsystems/PivotSubsystem.java [hunk 17]
```diff
@@ -6,54 +6,92 @@
...
-    pivotMotor = new CANSparkMax(IntakeConstants.PIVOT_MOTOR, MotorType.kBrushless);
-    pivotMotor.restoreFactoryDefaults();
-    pivotMotor.setSmartCurrentLimit(IntakeConstants.MOTOR_CURRENT_LIMIT);
-    pivotMotor.setIdleMode(CANSparkMax.IdleMode.kBrake);
-    pivotMotor.burnFlash();
+    super(
+        // The controller that the command will use
+        new PIDController(ManipulatorConstants.PIVOT_kP, ManipulatorConstants.PIVOT_kI, ManipulatorConstants.PIVOT_kD));
+
+    getController().setTolerance(-2);
```
Comment (minhd-vu) [inline src/main/java/frc/robot/subsystems/PivotSubsystem.java]: this should be a variable

**Review by minhd-vu — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/subsystems/PivotPIDSub.java [hunk 18]
```diff
@@ -0,0 +1,67 @@
...
+    pivotMotorL.burnFlash();
+
+    SmartDashboard.putNumber("PIVOT ENCODER", getMeasurement());
+
+  }
+
+  @Override
+  public void useOutput(double output, TrapezoidProfile.State setpoint) {
+    // Use the output (and optionally the setpoint) here
+    double feedforward = pivotfeedforward.calculate(setpoint.position, setpoint.velocity);
```
Comment (Shiv-Shah) [inline src/main/java/frc/robot/subsystems/PivotPIDSub.java:56]: Why no acceleration which is 0.1? https://docs.wpilib.org/en/stable/docs/software/advanced-controls/controllers/feedforward.html

**Pull request closed without merging**